dot/core/messages_test.go

```diff
@@ -2,3 +2,265 @@
 // SPDX-License-Identifier: LGPL-3.0-only
 
 package core
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/ChainSafe/gossamer/dot/network"
+	"github.com/ChainSafe/gossamer/dot/peerset"
+	"github.com/ChainSafe/gossamer/dot/types"
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/lib/runtime"
+	mocksruntime "github.com/ChainSafe/gossamer/lib/runtime/mocks"
+	"github.com/ChainSafe/gossamer/lib/runtime/storage"
+	"github.com/ChainSafe/gossamer/lib/transaction"
+
+	"github.com/golang/mock/gomock"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/stretchr/testify/assert"
+)
+
+var (
+	errBestHeader = errors.New("best header error")
+	errGetRuntime = errors.New("get runtime error")
+	errTrieState  = errors.New("trie state error")
+)
+
+// TODO use new test format
+
+func TestService_TransactionsCount(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mockTxnStateEmpty := NewMockTransactionState(ctrl)
+	mockTxnState := NewMockTransactionState(ctrl)
+
+	txs := make([]*transaction.ValidTransaction, 2)
+
+	mockTxnStateEmpty.EXPECT().PendingInPool().Return([]*transaction.ValidTransaction{})
+	mockTxnState.EXPECT().PendingInPool().Return(txs)
+
+	tests := []struct {
+		name    string
+		service *Service
+		exp     int
+	}{
+		{
+			name:    "empty",
+			service: &Service{transactionState: mockTxnStateEmpty},
+			exp:     0,
+		},
+		{
+			name:    "not empty",
+			service: &Service{transactionState: mockTxnState},
+			exp:     2,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			s := tt.service
+			res := s.TransactionsCount()
+			assert.Equal(t, tt.exp, res)
+		})
+	}
+}
+
+func TestServiceHandleTransactionMessage(t *testing.T) {
```

Comment:
Do you think it would be possible to define these mocks in the subtests using test case parameters instead?

[Example](https://github.com/ChainSafe/gossamer/blob/e1f7f96cb8cb7e82156f7333b9a0704ab8151fd2/internal/trie/node/header_test.go#L110-L121)

That would make it easier to follow how the mock is configured by just looking up the test case fields.

Reply:
I can certainly try!

Reply:
@qdm12 Maybe I'm not totally understanding, but I think this might be more tricky in my case since the different test cases have different mock calls. From what I can tell in this example each case calls the Write function from write mock, but if the mock methods called are variable per test case is this still possible/reasonable to do?

Reply:
You can have a `callXXX bool` field in the test case to check in the subtest to expect another mock call or not. It looks there is a lot of repetition so maybe try doing it for this test only to see how it is? 🤔

Reply:
okay yeah ill give it a try and see how it goes